volatility3/framework/symbols/linux/__init__.py

```diff
@@ -121,22 +121,34 @@ def get_path_mnt(cls, task, mnt) -> str:
         return cls.do_get_path(rdentry, rmnt, dentry, vfsmnt)
 
     @classmethod
-    def do_get_path(cls, rdentry, rmnt, dentry, vfsmnt) -> Union[None, str]:
+    def do_get_path(cls, rdentry, rmnt, dentry, vfsmnt) -> str:
         """Returns a pathname of the mount point or file
         It mimics the Linux kernel prepend_path function.
 
         Args:
             rdentry (dentry *): A pointer to the root dentry
             rmnt (vfsmount *): A pointer to the root vfsmount
             dentry (dentry *): A pointer to the dentry
-            vfsmnt (vfsmount *): A pointer to the vfsmount
+            vfsmnt (vfsmount/vfsmount *): A vfsmount object (kernels >= 3.3) or a
+                vfsmount pointer (kernels < 3.3)
 
         Returns:
             str: Pathname of the mount point or file
         """
 
+        if not (rdentry and rdentry.is_readable() and rmnt and rmnt.is_readable()):
+            return ""
+
+        if isinstance(vfsmnt, objects.Pointer) and not (rmnt and rmnt.is_readable()):
```

Comment:
I'd probably rewrite this as:

`if (rmt and rmnt.is_readable()):
  if not rdentry...:
    return None/""
else:
  if ininstance(vfsmnt...:
    return None/""
  ```
  
  It's fine as is, just reduces the complexity of the if clauses which should make it easier to read/parse.

Reply:
ups, that was actually a bug. The second `if` was meant to check the vfsmount pointer,  I just fixed it.